packages/dev/core/src/AudioV2/abstractAudio/spatialAttachers/spatialAudioCameraAttacher.ts

```diff
@@ -0,0 +1,107 @@
+import type { Camera } from "../../../Cameras/camera";
+import { Quaternion, Vector3 } from "../../../Maths/math.vector";
+import type { Nullable } from "../../../types";
+import type { ISpatialAudioNode } from "./abstractSpatialAudioAttacher";
+import { _AbstractSpatialAudioAttacher } from "./abstractSpatialAudioAttacher";
+import type { SpatialAudioAttachmentType } from "./spatialAudioAttacher";
+
+const TempQuaternion = Quaternion.Identity();
+
+/** @internal */
+export async function _CreateSpatialAudioCameraAttacherAsync(
+    camera: Camera,
+    spatialAudioNode: ISpatialAudioNode,
+    attachmentType: SpatialAudioAttachmentType,
+    minUpdateTime: number
+): Promise<_AbstractSpatialAudioAttacher> {
+    return new _SpatialAudioCameraAttacher(camera, spatialAudioNode, attachmentType, minUpdateTime);
+}
+
+/** @internal */
+export class _SpatialAudioCameraAttacher extends _AbstractSpatialAudioAttacher {
+    protected _camera: Nullable<Camera> = null;
+
+    /** @internal */
+    public constructor(camera: Camera, spatialAudioNode: ISpatialAudioNode, attachmentType: SpatialAudioAttachmentType, minUpdateTime: number) {
+        super(spatialAudioNode, attachmentType, minUpdateTime);
+
+        this.camera = camera;
+    }
+
+    /** @internal */
+    public set camera(camera: Nullable<Camera>) {
+        if (this._camera === this.camera) {
+            return;
+        }
+
+        this._clearCamera();
+
+        this._camera = camera;
+
+        this._setScene(this._camera?.getScene() ?? null);
+
+        this._isDirty = true;
+
+        this.update();
+    }
+
+    protected get _attachedPosition(): Vector3 {
+        return this._camera?.globalPosition ?? Vector3.ZeroReadOnly;
+    }
+
+    protected get _attachedRotationQuaternion(): Quaternion {
+        if (!this._camera) {
+            TempQuaternion.copyFromFloats(0, 0, 0, 1);
+            return TempQuaternion;
+        }
+
+        this._camera.getWorldMatrix().decompose(undefined, TempQuaternion, undefined);
```

Comment:
won't that be a problem to share the same quaternion among several entities? what if I have 2 cameras or 2 meshes with sound? they will overwrite their quaternion value

Reply:
I don't think so. A copy is made when setting the result on the spatial audio node:

https://github.com/BabylonJS/Babylon.js/blob/38b39538e67a49396bac8b0a6f8e49782b70b325/packages/dev/core/src/AudioV2/webAudio/subNodes/spatialWebAudioSubNode.ts#L150-L151

We could make another copy to be safe, but since this is only used internally and it's copied in the end I think it's ok to just return `TempQuaternion` here.